#!/usr/bin/env python
PACKAGE = "reassemble_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 200, 0, 500)
# gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 800, 0, 1000)
gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 500, 0, 1000)
# gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 10, 0, 30)
# gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 75, 0, 100)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 50, 0, 100)
gen.add("nullspace_damping", double_t, 0, "Damping of the joint space daming controller (the desired configuration is the one at startup)", 0.1, 0, 1)
gen.add("damping_ratio", double_t, 0, "Cartesian damping ratio", 1, 0, 2)
gen.add("manipulability_scale", double_t, 0, "P gain for the manipulability torques", 10, 0, 20)
gen.add("mani_vs_damping", double_t, 0, "P gain for the manipulability torques", 0.8, 0, 1)
gen.add("joint_limit_cont", double_t, 0, "how much the joint limits contribut to the torques", 10, 0, 20)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param"))
